video/out/cocoa_cb_common.swift

```diff
@@ -458,6 +456,26 @@ class CocoaCB: NSObject {
                         ccb.window?.toggleFullScreen(nil)
                     }
                 }
+                if opt! == UnsafeMutableRawPointer(&mpv.optsPtr.pointee.ontop) {
+                    DispatchQueue.main.async {
+                        ccb.window?.setOnTop(Bool(mpv.opts.ontop), Int(mpv.opts.ontop_level))
+                    }
+                }
+                if opt! == UnsafeMutableRawPointer(&mpv.optsPtr.pointee.keepaspect_window) {
+                    DispatchQueue.main.async {
+                        ccb.window?.keepAspect = Bool(mpv.opts.keepaspect_window)
+                    }
+                }
+                if opt! == UnsafeMutableRawPointer(&mpv.optsPtr.pointee.window_minimized) {
+                    DispatchQueue.main.async {
+                        ccb.window?.setMinimized(Bool(mpv.opts.window_minimized))
+                    }
+                }
+                if opt! == UnsafeMutableRawPointer(&mpv.optsPtr.pointee.window_maximized) {
+                    DispatchQueue.main.async {
+                        ccb.window?.setMaximized(Bool(mpv.opts.window_maximized))
+                    }
+                }
```

Comment:
You could create the m_config_cache directly on your swift GUI thread,instead of putting every single thing through the dispatch queue separately. 

Handling when to update the cache would be slightly trickier, but even then you could just use the VOCTRL and dispatch just the VOCTRL. It would also be possible to get a more "direct" notification (through wakeup callback), but as I said that might be trickier and would require exactly how the wakeup callback works and making it interact well with the GUI thread.